File: tools/RLMResearchPlannerPWA/src/resource-format.js
export function formatResourceAmount(amount, mode = "exact", locale = "ja-JP") {
  const value = Math.trunc(Number(amount) || 0);
  if (mode !== "short" || Math.abs(value) < 1_000) return value.toLocaleString(locale);
  for (const [divisor, suffix] of [[1e12, "T"], [1e9, "B"], [1e6, "M"], [1e3, "K"]]) {
    if (Math.abs(value) >= divisor) {
      return `${Number((value / divisor).toFixed(2))}${suffix}`;
    }
  }
  return value.toLocaleString(locale);
}

File: src/rlm_research_planner/services/resource_format.py
from __future__ import annotations


def format_resource_amount(amount: int, mode: str = "exact") -> str:
    value = int(amount)
    if mode != "short" or abs(value) < 1_000:
        return f"{value:,}"

    for divisor, suffix in (
        (1_000_000_000_000, "T"),
        (1_000_000_000, "B"),
        (1_000_000, "M"),
        (1_000, "K"),
    ):
        if abs(value) >= divisor:
            hundredths = abs(value) * 100 // divisor
            shortened = hundredths / 100
            if value < 0:
                shortened = -shortened
            return f"{shortened:.2f}".rstrip("0").rstrip(".") + suffix
    return f"{value:,}"
